### DISCUSSION TITLE
be good to have a few warning messages

### DISCUSSION THREAD

**Discussion by dbenielli** [archimede-institut/GINsim#4] [Announcements]
# Bug correction

- BUG 3 It would be good to have a few warning messages

1. asking for confirmation when deleting a component e.g. « delete a component will lead to

incorrect rules of its targets. Do you want to proceed ? »

2. asking for confirmation when deleting an interaction e.g. « delete an interaction will lead to

an incorrect rule of its target. Do you want to proceed ? »

3. indicating that reversing a model is only possible for Boolean models (in this case, a

message is displayed in the console)

4.  others ?

Note that we should avoid having to confirm too many actions !

**Comment by dbenielli** [comment]
done in 186ffde5a387b8873102aea6a17e3b933c6fda9f commit add delete message
1 parent 
[c82ace1](https://github.com/GINsim/GINsim/commit/c82ace1634ef49043ee343a609c592a4783e01d1)
 commit 
186ffde

**Comment by chaouiya** [comment]
I checked quickly and it seems to work well! I would only suggest to change the warning message as follow
1) when deleting a component: "Deleting a component may lead to incorrect functions of its targets. Do you want to proceed?"
2) when deleting an interaction: "Deleting an interaction may lead to an incorrect function of its targets.  Do you want to proceed?"
with labels of the buttons being Yes No
Thanks!
Claudine

**Reply by dbenielli** [reply]
Remarks 1 and 2 from Claudine implemented in commit improve delete messages 8161bf343661ac2ba3e349736c8a9cf1ffbf8fbd
[dbenielli](https://github.com/dbenielli)pushed 1 commit to [main](https://github.com/archimede-institut/GINsim/activity?ref=main) • 186ffde…8161bf3 • 12 minutes ago
